### ISSUE TITLE
subword `#` should be an option.

### ISSUE THREAD

**Issue by FFengIll** [skeskinen/bert.cpp#33]
For bert, there are many models use `#` for subword symbol, but not all.
Some popular bert-based models defined their own subword symbol.

For example, in `e5` the symbol is `▁`.

```
>>> a = '▁'
>>> a.encode('utf-8')
b'\xe2\x96\x81'
```

**Comment by FFengIll** [comment]
Furthermore, there is no such rule to force use `#`.

**Comment by FFengIll** [comment]
In model, the substr symbol always be called as `replacement` or `continuing_subword_prefix`.
Actually, it will show in `tokenizer.json`.

**Comment by FFengIll** [comment]
@skeskinen  no, #31  only make vocab not necessary (because it maybe missing).

This issue is another problem for subwords ( I found this since I meet too many unknown token when using e5).

bellow is some token samples in bert-based model.

in m3e, subword is ## like many bert model.
```
"##a": 8139,
"03": 8140,
"09": 8141,
"08": 8142,
"28": 8143,
"##2": 8144,
```

in e5, subword is `▁` since they trained a new tokenizer (bellow is part copy from tokenizer.json)
```
      [
        "▁si",
        -7.355116367340088
      ],
      [
        "▁ja",
        -7.370460510253906
      ],
      [
        "▁za",
        -7.37307596206665
      ],
      [
        "▁v",
        -7.385393142700195
      ],

```

**Comment by FFengIll** [comment]
For now, I do not have a good idea for this issue, so I do not implement a PR for it.
Maybe we need to more research and discuss.

**Comment by cgisky1980** [comment]
> For now, I do not have a good idea for this issue, so I do not implement a PR for it. Maybe we need to more research and discuss.

加油，需要跨平台的中英文向量化~
